src/components/CallTile.vue

```diff
@@ -98,63 +102,219 @@ export default {
+      .off("track-started", (p) => {
+        if (!p?.participant) return;
+        const tracks = this.getParticipantTracks(p.participant);
+        try {
+          this.updateMedia(p.participant.session_id, tracks);
+        } catch (e) {
+          console.warn(e);
+        }
+      })
+      .off("track-stopped", (p) => {
+        if (!p?.participant) return;
+        const tracks = this.getParticipantTracks(p.participant);
+        try {
+          this.updateMedia(p.participant.session_id, tracks);
+        } catch (e) {
+          console.warn(e);
+        }
+      });
   },
   methods: {
-    /**
-     * This is called any time a participant update registers.
-     * In large calls, this should be optimized to avoid re-renders.
-     * For example, track-started and track-stopped can be used
-     * to register only video/audio/screen track changes.
-     */
-    updateParticpants(e) {
-      console.log("[EVENT] ", e);
-      if (!this.callObject) return;
+    updateMedia(participantID, newTracks) {
+      // Get the video tag.
+      let videoTile = document.getElementById(
+        participantID
+      );
+      if (!videoTile) {
+        const videoCall = this.$refs.videoCall;
+        
+        const newVideoTile = document.createElement("video");
+        newVideoTile.id = participantID;
+        videoCall.appendChild(newVideoTile);
+        videoTile = newVideoTile;
+      }
 
-      const p = this.callObject.participants();
-      this.count = Object.values(p).length;
-      this.participants = Object.values(p);
+      const video = videoTile;
 
-      const screen = this.participants.filter((p) => p.screenVideoTrack);
-      if (screen?.length && !this.screen) {
-        console.log("[SCREEN]", screen);
-        this.screen = screen[0];
-      } else if (!screen?.length && this.screen) {
-        this.screen = null;
+      // Get existing MediaStream from the video tag source object.
+      const existingStream = video.srcObject;
+
+      const newVideo = newTracks.videoTrack;
+      const newAudio = newTracks.audioTrack;
+
+      // If there is no existing stream or it contains no tracks,
+      // Just create a new media stream using our new tracks.
+      // This will happen if this is the first time we're
+      // setting the tracks.
+      if (!existingStream || existingStream.getTracks().length === 0) {
+        const tracks = [];
+        if (newVideo) tracks.push(newVideo);
+        if (newAudio) tracks.push(newAudio);
+        const newStream = new MediaStream(tracks);
+        video.srcObject = newStream;
+        video.playsInline = true;
+        video.autoplay = true;
+        video.muted = true;
+        this.playMedia(video);
+        return;
+      }
+
+      // This boolean will define whether we play the video element again
+      // This should be `true` if any of the tracks have changed.
+      let needsPlay = false;
+      needsPlay = this.refreshAudioTrack(existingStream, newAudio);
```

Comment:
```suggestion
      let needsPlay = this.refreshAudioTrack(existingStream, newAudio);
```

Or is there a good reason for the reassign?